Source/EasyNetQ/Internals/NoResult.cs

```diff
@@ -0,0 +1,16 @@
+namespace EasyNetQ.Internals;
+
+/// <summary>
+///     This is an internal API that supports the EasyNetQ infrastructure and not subject to
+///     the same compatibility as public APIs. It may be changed or removed without notice in
+///     any release. You should only use it directly in your code with extreme caution and knowing that
+///     doing so can result in application failures when updating to a new EasyNetQ release.
+/// </summary>
+public record NoResult
```

Comment:
Why so much "pubinternal" APIs ? Why not convert them into **real** internal APIs and add nessessary InternalVisibleTo properties?

Reply:
To write extensions easily I suppose.

Reply:
😕 So these APIs are public. It is strange to position them to write extensions (like any other external code), while refusing any guarantees of stability. Whatever.w

Reply:
It is not a topic of this PR, please open a new discussion,